OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnAmmo.cs

```diff
@@ -0,0 +1,73 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Applies a condition based on the status of an AmmoPool.")]
+	public class GrantConditionOnAmmoInfo : ITraitInfo, Requires<AmmoPoolInfo>
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("Condition to grant.")]
+		public readonly string Condition = null;
+
+		[FieldLoader.Require]
+		[Desc("Expression to evaluate. Use the variables \"amount\" for the current ammo count and \"capacity\" for the capacity.")]
+		public readonly BooleanExpression Expression;
+
+		[Desc("Name of the AmmoPools.")]
+		public readonly string AmmoPoolName;
+
+		public object Create(ActorInitializer init) { return new GrantConditionOnAmmo(init.Self, this); }
+	}
+
+	public class GrantConditionOnAmmo : INotifyCreated, ITick
+	{
+		readonly GrantConditionOnAmmoInfo info;
+
+		ConditionManager conditionManager;
+		int conditionToken = ConditionManager.InvalidConditionToken;
+		AmmoPool[] ammoPools;
+
+		public GrantConditionOnAmmo(Actor self, GrantConditionOnAmmoInfo info)
+		{
+			this.info = info;
+		}
+
+		void INotifyCreated.Created(Actor self)
+		{
+			conditionManager = self.TraitOrDefault<ConditionManager>();
+			ammoPools = self.TraitsImplementing<AmmoPool>().Where(x => x.Info.Name == info.AmmoPoolName).ToArray();
+		}
+
+		public void Tick(Actor self)
+		{
+			var variables = ReadOnlyDictionary.AsReadOnly(new Dictionary<string, int>()
+				{
+					{ "amount", ammoPools.Sum(x => x.GetAmmoCount()) },
```

Comment:
Can you please also add a comment `// TODO: Update these to account for disabled traits once AmmoPool is made conditional`